Automattic/pocket-casts-android
# Whats new: Autoplay

## Description
This PR:
1. Adds a what's new dialog announcing the autoplay feature with new styling for the dialog. As part of this the old what's new dialog code is removed.
2. Removes the autoplay feature flag because we've decided to release this with `7.43`. **That means this PR is targeting `release/7.43`.**

The figma designs requested a pop-up dialog for this, but we've been using a bottom sheet everywhere else in the app. In this PR, I'm making a bottom sheet look like a pop-up dialog by fading out the scrim before the "dialog" is dismissed. If we stick with using dialogs, we may want to do a more complete switch, but for now I thought that this worked pretty well and helped to minimize the number of changes we need.

The language we're using in the What's new" dialog ("will autoplay episodes from your current list, if your Up Next queue is empty") seems misleading to me because it indicates that autoplay will play a new episode if I listen to all the episodes in my up next queue. In fact, autoplay only works if the Up Next queue is empty when you start playing the episode _and_ the Up Next queue stays empty the entire time you listen to the episode. 

I don't have any great ideas for how to communicate that in a short blurb though. If you have any suggestions for how to improve that, let me know because I am concerned that our users will be confused when they finish their up next queue and autoplay doesn't do anything. internal ref: p1689101648157169/1689091788.481629-slack-C02ATC80MUM

Related iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/956

## Testing Instructions

### 1. Verify What's New is shown when appropriate
1. Download a 7.19.2 build from the Google Play Store (basically any build before folders was released in 7.20), install it, and start the app. Observe there is no announcement pop-up/dialog
2. Download a 7.42 build from the Google Play Store, install it, and start the app. Dismiss the onboarding flow and observe the old "What's New" dialog is displayed.
4. Change the code in this PR [to force the build number to be 9115](https://gist.github.com/mchowning/2ce380ac8f04c5085b89f7376ea849a8), and install a release build.
5. Observe that the new What's New dialog is shown announcing the autoplay feature. Note that the animation on this is smooth on release builds, but really choppy on debug builds.

### 2. Test What's New dialog
- Verify that tapping on the "Enable it now" button (1) closes the dialog, (2) Opens the general settings, (3) scrolls to the bottom of that screen, and (4) flashes the autoplay setting
- Verify that tapping either "Maybe later" or somewhere outside the dialog closes the dialog.

### 3. Confirm removal of feature flag
- Verify that on debug builds there is no longer a feature flag option in the settings and the autoplay setting is visible.
- Verify that on release builds the autoplay setting is visible

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/4656348/30277b60-dc30-4ab2-b477-08f638e1c68c



||||||
| :---: | :---: | :---: | :---: | :---: |
| Light | Dark | Rosé | Indigo | Extra Dark |
| ![Screenshot 2023-07-11 at 2 18 14 PM](https://github.com/Automattic/pocket-casts-android/assets/4656348/e473579a-cf1f-4134-acf6-201ebf893c5a) | ![Screenshot 2023-07-11 at 2 19 32 PM](https://github.com/Automattic/pocket-casts-android/assets/4656348/8cda3471-a7ef-4d72-a230-1ec13cd30e14) | ![Screenshot 2023-07-11 at 2 19 48 PM](https://github.com/Automattic/pocket-casts-android/assets/4656348/b5970b40-c706-402b-bc8d-b835d973ebd6) | ![Screenshot 2023-07-11 at 2 20 03 PM](https://github.com/Automattic/pocket-casts-android/assets/4656348/85b59953-d23f-4f8b-bc37-b121e543b775) | ![Screenshot 2023-07-11 at 2 20 21 PM](https://github.com/Automattic/pocket-casts-android/assets/4656348/af234f36-3507-4eab-b72c-19cd36f7b5a7) |
| Dark Contrast | Light Contrast | Electricity | Classic | Radioactivity |
| ![Screenshot 2023-07-11 at 2 21 18 PM](https://github.com/Automattic/pocket-casts-android/assets/4656348/de3a7172-fe58-4740-ab6c-99c0f453e1fa) | ![Screenshot 2023-07-11 at 2 21 30 PM](https://github.com/Automattic/pocket-casts-android/assets/4656348/8d32956b-a610-4eaf-bd24-297edf86d906) | ![Screenshot 2023-07-11 at 2 21 43 PM](https://github.com/Automattic/pocket-casts-android/assets/4656348/47cc0519-bb85-482b-82f2-f84f50d2f93e) | ![Screenshot 2023-07-11 at 2 22 05 PM](https://github.com/Automattic/pocket-casts-android/assets/4656348/1f467be8-4092-467e-8ed5-3b7be8eba4d2) | ![Screenshot 2023-07-11 at 2 22 17 PM](https://github.com/Automattic/pocket-casts-android/assets/4656348/0854be25-a953-4070-a0a3-f3390ae2709b) |




## Checklist
 
#### I have tested any UI changes...
